src/tools/registry/gen_inc.c

```diff
@@ -645,7 +645,11 @@ int parse_namelist_records_from_registry(ezxml_t registry)/*{{{*/
 		}
 
 		fortprintf(fd, "      if (dminfo %% my_proc_id == IO_NODE) then\n");
+		fortprintf(fd, "! Rewinding before each read leads to errors when the code is built with\n");
+		fortprintf(fd, "! the NAG Fortran compiler. If building with NAG, be kind and don't rewind.\n");
```

Comment:
😒 ...

I'm guessing the NAG compiler rewinds automatically once read automatically reaches the end of a record?

Reply:
No, it appears that the NAG compiler expects the file to be positioned at the beginning of the next namelist group (possibly preceded by comments); if we rewind the file, this isn't true. For example, if we have the following `namelist.atmosphere` file:
```
&foo
    a = 2.71828
/
&blah
    i = 42
/
```
rewinding before trying to read the `&blah` group results in an error.